### ISSUE TITLE
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'

### ISSUE THREAD

**Issue by patel-zeel** [tom-andersson/deepsensor#51]
Hi,

When I try to install and import `deepsensor` for Python<3.10, it throws me the error below:

```py
from deepsensor.data.loader import TaskLoader
```
```
class TaskLoader:
     17     config_fname = "task_loader_config.json"
     19     def __init__(
     20         self,
---> 21         task_loader_ID: str | None = None,

TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

It happens because Python<3.10 is not converting `str | None` to `types.UnionType` but instead throws an error.

Does `deepsensor` support Python>=3.10 only from now on? I am thinking of a scenario (which actually occurred to me) where this becomes a challenge: 
- Say I have a Python<3.10 environment set up with `torch` and several other dependencies needed for my project. Currently, I am unable to install `deepsensor` in the same environment due to this breaking change. I have to set up a new environment with Python>=3.10 and reinstall `torch` as well as other N libraries to make it work for a project. 
- This may be a common problem with any breaking changes in Python. In that case, this issue will help users to know the exact cause of this error and take appropriate steps (install Python>=3.10) to solve it. 

Would it be helpful to mention the minimum supported Python version in README somewhere around `pip install deepsensor` Instructions?

**Comment by tom-andersson** [comment]
Hi @patel-zeel, thank you for catching this. It wasn't intentional to break Python < 3.10; I didn't realise the `|` syntax was Python >= 3.10 only. Sorry!

Currently, the minimum version in `setup.cfg` is set to `3.8`: https://github.com/tom-andersson/deepsensor/blob/ca079cc8e6e12be3719caf7dd8ec91ecd701a8ae/setup.cfg#L24

However, the continuous integration was only testing `3.11` to reduce compute, which is why I missed this bug: https://github.com/tom-andersson/deepsensor/blob/ca079cc8e6e12be3719caf7dd8ec91ecd701a8ae/.github/workflows/tests.yml#L13

I've fixed the Python<3.10 incompatible syntax in `main` in https://github.com/tom-andersson/deepsensor/commit/7793a71c21b0646dfa80344c5524547bf477ca65 and that fixes the bug locally for me. And I've now added Python 3.8 to the CI. Will close this if test pass.

At some point we should discuss the pros/cons of different oldest Python versions from both a development and user point of view.

**Comment by patel-zeel** [comment]
> I've fixed the Python<3.10 incompatible syntax in main in https://github.com/tom-andersson/deepsensor/commit/7793a71c21b0646dfa80344c5524547bf477ca65 and that fixes the bug locally for me. And I've now added Python 3.8 to the CI. Will close this if test pass.

Perfect, @tom-andersson! Thank you. Now it works without any errors.

> At some point we should discuss the pros/cons of different oldest Python versions from both a development and user point of view.

Absolutely. If we think about the potential users of `deepsensor`, it gets more contextualized. My initial thoughts are leaning towards extended compatibility over syntactic convenience (such as `str | None` < `Union[str, None]`) given that many people may use `deepsensor` for purposes like scaling+inverse-scaling data, creating context-target splits for their models etc. Some old libraries may not support Python beyond a limit if `deepsensor` becomes limited to certain Python versions. There could be some Python-version-specific End-of-life releases also which says something like, "beyond this point, Python=3.x support is dropped" and a table in README shows latest `deepsensor` version that supports Python<=3.x to easily navigate the users.

**Comment by tom-andersson** [comment]
Hey @patel-zeel, I completely agree with your thoughts above. No compelling reason to constrain to recent Python versions yet, but if this becomes necessary, we should follow best practices in the documentation (as you suggest).
